src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java

```diff
@@ -1061,14 +1059,11 @@ public Boolean visitType(TypeElement e, Void p) {
                 }
                 Element enclosing = e.getEnclosingElement();
                 if (enclosing != null) {
-                    switch(enclosing.getKind()) {
-                        case PACKAGE:
-                            return specifiedPackageElements.contains((PackageElement)enclosing);
-                        case CLASS: case INTERFACE: case ENUM: case ANNOTATION_TYPE:
-                            return visit((TypeElement) enclosing);
-                        default:
-                            throw new AssertionError("unknown element: " + enclosing);
-                    }
+                    return switch (enclosing.getKind()) {
+                        case CLASS, ENUM, RECORD, INTERFACE, ANNOTATION_TYPE -> visit(enclosing);
```

Comment:
Whoa! `RECORD` was missing. Does it make sense to accompany this PR with a small test that crashes javadoc with a type nested in a non-included record?

Reply:
I've reviewed all uses of `ENUM`. All other places in javadoc look OK for having a sibling case for `RECORD`.

Filed [JDK-8286205](https://bugs.openjdk.java.net/browse/JDK-8286205) for more tests.